Lib/idlelib/idle_test/test_calltips.py

```diff
@@ -46,25 +46,36 @@ def test_builtins(self):
 
         # Python class that inherits builtin methods
         class List(list): "List() doc"
+
         # Simulate builtin with no docstring for default tip test
         class SB:  __call__ = None
 
         def gtest(obj, out):
             self.assertEqual(signature(obj), out)
 
         if List.__doc__ is not None:
-            gtest(List, List.__doc__)
+            gtest(List, '(iterable=(), /)\n' + ct._argument_positional + '\n' +
+                  List.__doc__)
         gtest(list.__new__,
-               'Create and return a new object.  See help(type) for accurate signature.')
+               '(*args, **kwargs)\nCreate and return a new object.  See help(type) for accurate signature.')
         gtest(list.__init__,
+               '(self, /, *args, **kwargs)\n' + ct._argument_positional + '\n' +
                'Initialize self.  See help(type(self)) for accurate signature.')
-        append_doc =  "Append object to the end of the list."
-        gtest(list.append, append_doc)
-        gtest([].append, append_doc)
-        gtest(List.append, append_doc)
+        append_doc = ct._argument_positional + '\n' + "Append object to the end of the list."
+        gtest(list.append, '(self, object, /)\n' + append_doc)
+        gtest(List.append, '(self, object, /)\n' + append_doc)
+        gtest([].append, '(object, /)\n' + append_doc)
 
         gtest(types.MethodType, "method(function, instance)")
         gtest(SB(), default_tip)
+        import re
+        p = re.compile('')
+        gtest(re.sub, '''(pattern, repl, string, count=0, flags=0)\nReturn the string obtained by replacing the leftmost
+non-overlapping occurrences of the pattern in string by the
+replacement repl.  repl can be either a string or a callable;
+if a string, backslash escapes in it are processed.  If it is
+a callable, it's passed the match object and must return''')
```

Comment:
I would prefer that this be truncated also.  We allowed up to 5 docstring lines for builtins to get the signature (bytes needs 5), but if we already have signature, only 1 should be allowed, as below.  However, after checking the code and the docstring, I decided that the source of the problem is that the re.sub docstring is defective in not having an overly long summary sentence and then not having a blank line after the first '.', as it should.  We could check for '.', but not worth the effort now.  Instead I would like to add a comment so it is clear that this case was thought about.  This is a good additional test case.

`# Comment runs on because of long first sentence and missing blank thereafter.  Display as above.`